AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/Storage/SQLite/Statement+Model.swift

```diff
@@ -122,7 +122,10 @@ extension Statement: StatementModelConvertible {
             // For example, when the value is the `id` of the Blog, then the field.isPrimaryKey is satisfied.
             // Every association of the Blog, such as the has-many Post is populated with the List with
             // associatedId == blog's id. This way, the list of post can be lazily loaded later using the associated id.
-            if let id = modelValue as? String, field.isPrimaryKey {
+            if let id = modelValue as? String,
+               (field.name == ModelIdentifierFormat.Custom.sqlColumnName || // this is the `@@primaryKey` (CPK)
+                (schema.primaryKey.fields.count == 1 // or there's only one primary key (not composite key)
+                 && schema.primaryKey.indexOfField(named: field.name) != nil)) { // and this field is the primary key
```

Comment:
`isPrimaryKey` is deprecated and the new logic successfully returns true for use cases using CPK feature flag for models without a CPK use case, ie. Post/Comment using the default identifier. The problem is `isPrimaryKey` is return false, thus the lazy list was no longer being injected in the post model. This change is needed to create lazy lists and was a bug introduced with the CPK changes